src/main/java/hexgui/gui/HexGui.java

```diff
@@ -0,0 +1,2393 @@
+    Vector<Pair<String, String>> pairs = StringUtils.parseStringPairList(str);
+    m_guiboard.clearMarks();
+    for (int i = 0; i < pairs.size(); i++) {
+      HexPoint point = HexPoint.get(pairs.get(i).first);
+      String value = pairs.get(i).second;
+      m_guiboard.setText(point, value);
+    }
+    m_guiboard.repaint();
+  }
+
+  public void cbDisplayBookScores() {
+    if (!m_white.wasSuccess()) return;
+    String str = m_white.getResponse();
+    Vector<Pair<String, String>> pairs = StringUtils.parseStringPairList(str);
+    m_guiboard.clearMarks();
+    for (int i = 0; i < pairs.size(); i++) {
+      HexPoint point = HexPoint.get(pairs.get(i).first);
+      String value = pairs.get(i).second;
+      m_guiboard.setText(point, value);
+      if (i == 0) m_guiboard.setAlphaColor(point, Color.red);
+      else if (1 <= i && i <= 3) m_guiboard.setAlphaColor(point, Color.green);
+    }
+    m_guiboard.repaint();
+  }
+
+  public void cbEvalResist() {
+    if (!m_white.wasSuccess()) return;
+    String str = m_white.getResponse();
+    Vector<Pair<String, String>> pairs = StringUtils.parseStringPairList(str);
+    String res = "";
+    String rew = "";
+    String reb = "";
+    m_guiboard.clearMarks();
+    for (int i = 0; i < pairs.size(); i++) {
+      if (pairs.get(i).first.equals("res")) {
+        res = pairs.get(i).second;
+      } else if (pairs.get(i).first.equals("rew")) {
+        rew = pairs.get(i).second;
+      } else if (pairs.get(i).first.equals("reb")) {
+        reb = pairs.get(i).second;
+      } else {
+        HexPoint point = HexPoint.get(pairs.get(i).first);
+
+        String value = pairs.get(i).second;
+        m_guiboard.setText(point, value);
+      }
+    }
+    m_guiboard.repaint();
+    m_statusbar.setMessage("Resistance: " + res + " (" + rew + " - " + reb + ")");
+  }
+
+  public void cbEditParameters() {
+    if (!m_white.wasSuccess()) return;
+    String response = m_white.getResponse();
+    ParameterDialog.editParameters(
+        m_curAnalyzeCommand.getCommand(),
+        this,
+        "Edit Parameters",
+        response,
+        m_white,
+        m_messageDialogs);
+  }
+
+  public void cbSolveState() {
+    if (!m_white.wasSuccess()) return;
+    String response = m_white.getResponse();
+    m_statusbar.setMessage(format("Winning: {0}", response));
+  }
+
+  // ==================================================
+  // gfx commands
+  // ==================================================
+  public void guifx(String fx) {
+    System.out.println("gogui-gfx:\n'" + fx + "'");
+
+    if (fx.length() > 3 && fx.substring(0, 3).equals("uct")) guifx_uct(fx.substring(3));
+    else if (fx.length() > 2 && fx.substring(0, 2).equals("ab")) guifx_ab(fx.substring(2));
+    else if (fx.length() > 4 && fx.substring(0, 4).equals("dfpn")) guifx_dfpn(fx.substring(4));
+    else if (fx.length() > 6 && fx.substring(0, 6).equals("solver")) guifx_solver(fx.substring(6));
+  }
+
+  private void guifx_uct(String fx) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.